### ISSUE TITLE
iOS Background Location

### ISSUE THREAD

**Issue by Wackymax** [Baseflow/flutter-geolocator#948]
## 🚀 Feature Requests

Hi, would it be possible to add a way for us to tweak the background location setting of the iOS location manager?

### Contextualize the feature

```
AppleSettings(
        accuracy: LocationAccuracy.high,
        activityType: ActivityType.airborne,
        allowsBackgroundLocationUpdates: true,
        showsBackgroundLocationIndicator: true,
      )
```

Would set

```
if (@available(iOS 9.0, *)) {
                self.clLocationManager.allowsBackgroundLocationUpdates = enable;
            }
            if (@available(iOS 11.0, *)) {
                self.clLocationManager.showsBackgroundLocationIndicator = enable;
            }
```

### Describe the feature

Adding addition options for Apple Settings to configure the clLocationManager allowsBackgroundLocationUpdates and showsBackgroundLocationIndicator.

### Platforms affected (mark all that apply)
- [X] :iphone: iOS
- [ ] :robot: Android

**Comment by mvanbeusekom** [comment]
Hi @Wackymax, 

There is a way to do so but is not documented. You can enable the `allowsBackgroundLocationUpdates` property by adding the `location` key to the `UIBackGrounds` collection in your `ios/Runner/Info.plist` file. The easiest way to do this is by opening your project in Xcode (make sure you open the `ios` folder in your project) and add the Background Modes capability to your XCode project (`Project > Signing and Capabilities > "+ Capability"` button) and select "Location Updates".

The current implementation looks like this ([GeolocationHandler.m#L117-123(https://github.com/Baseflow/flutter-geolocator/blob/291ebf052ac562756fae67242d1db45114031a1c/geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m#L117)]:

```objc
+ (BOOL) shouldEnableBackgroundLocationUpdates {
    if (@available(iOS 9.0, *)) {
        return [[NSBundle.mainBundle objectForInfoDictionaryKey:@"UIBackgroundModes"] containsObject: @"location"];
    } else {
        return NO;
    }
}
```

and is called on [line 71](https://github.com/Baseflow/flutter-geolocator/blob/291ebf052ac562756fae67242d1db45114031a1c/geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m#L71)when you start listing to the stream returned from `getPositionStream()`:

```objc
#if TARGET_OS_IOS
  if (@available(iOS 9.0, macOS 11.0, *)) {
      locationManager.allowsBackgroundLocationUpdates = [GeolocationHandler shouldEnableBackgroundLocationUpdates];
  }
#endif
```

I hope this answers your question, I will close this issue for now. Feel free to leave a comment if you have additional questions.

**Comment by Wackymax** [comment]
Perfect. Thanks!
